laboratory/src/org/labkey/laboratory/assay/AssayHelper.java

```diff
@@ -136,7 +136,7 @@ public Map<String, Object> saveTemplate(User u, Container c, ExpProtocol protoco
             if (templateId == null)
             {
                 row = Table.insert(u, ti, row);
-                templateId = (Integer)row.get("rowid");
+                row.get("rowid");
```

Comment:
That's interesting how it did the refactor here. Assigning a value to templateId is unnecessary, which it found. However, leaving line 139 as-is should not happen. Line 139 can be removed.

Reply:
@labkey-jeckels : do you want me to make this edit here, or will you do these on your machine?

Reply:
> That's interesting how it did the refactor here. Assigning a value to templateId is unnecessary, which it found. However, leaving line 139 as-is should not happen. Line 139 can be removed.

It doesn't modify the right-hand side of the operation since it doesn't know if there are side-effects (in general).

Reply:
I removed the line completely. I agree with Nick's assessment of why the refactor made the edit it did. I've seen cases where it completely removed method calls that were part of unused assignment statements, but only in cases where it knew there was no side effect, like a trivial `get` method.